### PR TITLE
Plugin system is ready

### PR DESCRIPTION
**Pull request by RanMaoyi**
related https://github.com/nanomq/NanoNNG/pull/811
and windows is not supported now

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -207,6 +207,11 @@ if(ENABLE_PARQUET)
   add_definitions(-DSUPP_PARQUET)
 endif()
 
+if(ENABLE_PLUGIN)
```
Comment (JaylinYu) [inline CMakeLists.txt:210]: check platform

File: nanomq/CMakeLists.txt [hunk 2]
```diff
@@ -54,6 +54,14 @@ set(SOURCES
     apps/broker.c
 )
 
+if(NNG_ENABLE_PLUGIN)
+    if (NOT WIN32 AND NOT CYGWIN)
```
Comment (JaylinYu) [inline nanomq/CMakeLists.txt:58]: cool

**Review by JaylinYu — APPROVED** [review]
Thinking of how to apply plug-in to other place

File: CMakeLists.txt (hunk 1 above)
Comment (RanMaoyi) [reply CMakeLists.txt:210]: done, move the check in SUPP_PLUGIN to ENABLE_PLUGIN

**Review by OdyWayne — APPROVED** [review]

File: etc/nanomq_example.conf [hunk 3]
```diff
@@ -1199,3 +1199,9 @@ parquet {
 	# # Default: 5
 	file_count = 5
 }
+
+plugin {
+    libs = [{
+        path = "/path/to/plugin_user_property.so"
```
Comment (wanghaEMQ) [inline etc/nanomq_example.conf:1205]: Should add more comments

**Review by wanghaEMQ — APPROVED** [review]

File: nanomq/plugin/plugin_user_property.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+int cb(void *data)
+{
+	char **property = data;
+	if (property != NULL) {
+		property[0] = malloc(strlen("alpha") + 1);
+		strcpy(property[0], "alpha");
+		property[1] = malloc(strlen("beta") + 1);
+		strcpy(property[1], "beta");
+	}
```
Comment (wanghaEMQ) [inline nanomq/plugin/plugin_user_property.c:26]: The property seems not to be used.

Comment (RanMaoyi) [reply nanomq/plugin/plugin_user_property.c:26]: used in plugin_hook_call(HOOK_USER_PROPERTY, uproperty);

Comment (wanghaEMQ) [reply nanomq/plugin/plugin_user_property.c:26]: Yes. you are right.

**Review by lee-emqx — APPROVED** [review]

File: etc/nanomq_example.conf (hunk 3 above)
Comment (JaylinYu) [reply etc/nanomq_example.conf:1205]: agree

**Pull request merged**